docs/appendices/release-notes/unreleased.rst

```diff
@@ -133,3 +133,6 @@ Fixes
 - Fixed an issue that caused queries reading values of type ``BIT`` to return a
   wrong result if the query contains a ``WHERE`` clause ``pk_col = ?``
   condition.
+
+- Fixed an issue that caused ignoring ``NOT NULL`` constraint on ``GEO_SHAPE``
```

Comment:
Not sure whether losing flag in `DocIndexMetadata` has more serious consequnce: insert(null) throws "must not be null" exception as expected due to early validation.